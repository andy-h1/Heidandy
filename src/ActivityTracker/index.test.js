import React from 'react';
import axiosMock from 'axios';
import userEvent from '@testing-library/user-event';
import { cleanup, render, screen, waitFor } from '@testing-library/react';
import { ActivityTracker } from '.';

afterEach(cleanup);
jest.mock('axios');

const activityData = [
  {
    complete: false,
    cost: 5000.0,
    description: 'Destroying scrubs',
    id: 4,
    location: 'Haven',
    name: 'Valorant',
  },
  {
    complete: false,
    cost: 100.0,
    description: 'Getting destroyed',
    id: 1,
    location: 'Summoners',
    name: 'League',
  },
];

const defaultProps = {
  fetchData: () => null,
};

describe('Activity Tracker', () => {
  it('should render', () => {
    const { container } = render(<ActivityTracker {...defaultProps} activities={activityData} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should delete the activity after button has been clicked', async () => {
    // render the component
    render(<ActivityTracker {...defaultProps} activities={activityData} />);

    // do stuff as a user
    userEvent.click(screen.getAllByText('Delete')[0]);

    // check that API was called
    await waitFor(() => screen.getByText('Deleting data...'));
    expect(axiosMock.delete).toHaveBeenCalledWith(`https://heidan-api.herokuapp.com/activity/${activityData[0].id}`);
  });

  it('should show error message if API is down', async () => {
    axiosMock.delete.mockRejectedValue({ error: '404' });
    render(<ActivityTracker {...defaultProps} activities={activityData} />);

    userEvent.click(screen.getAllByText('Delete')[0]);

    await waitFor(() => screen.getByText(`Sorry there was an error deleting ${activityData[0].name}`));
  });

  it('should render done if complete is true', () => {
    const completedActivity = [
      {
        complete: true,
        cost: 100.0,
        description: 'Getting destroyed',
        id: 1,
        location: 'Summoners',
        name: 'League',
      },
    ];

    render(<ActivityTracker {...defaultProps} activities={completedActivity} />);

    screen.getByText('Complete: done');
  });
});
