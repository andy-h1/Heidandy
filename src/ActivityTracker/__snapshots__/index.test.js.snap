// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Tracker should render 1`] = `
<div
  data-testid="activityTracker"
>
  <ul>
    <li>
      <p>
        Activity name: 
        Valorant
      </p>
      <p>
        Description: 
        Destroying scrubs
      </p>
      <p>
        Location: 
        Haven
      </p>
      <p>
        Cost: 
        5000
      </p>
      <p>
        Complete: 
        not done
      </p>
      <button
        class="sc-bdfBwQ bNUBxi"
        type="button"
      >
        Delete
      </button>
    </li>
    <li>
      <p>
        Activity name: 
        League
      </p>
      <p>
        Description: 
        Getting destroyed
      </p>
      <p>
        Location: 
        Summoners
      </p>
      <p>
        Cost: 
        100
      </p>
      <p>
        Complete: 
        not done
      </p>
      <button
        class="sc-bdfBwQ bNUBxi"
        type="button"
      >
        Delete
      </button>
    </li>
  </ul>
</div>
`;
